src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -449,8 +449,8 @@ subroutine diabatic(u, v, h, tv, Hml, fluxes, visc, ADp, CDp, dt, Time_end, &
     ! These stochastic perturbations do not conserve heat, salt or mass.
     do k=1,nz ; do j=js,je ; do i=is,ie
       h(i,j,k) = max(h_in(i,j,k) + (h(i,j,k)-h_in(i,j,k)) * stoch_CS%sppt_wts(i,j), GV%Angstrom_H)
-      tv%T(i,j,k) = t_in(i,j,k) + (tv%T(i,j,k)-t_in(i,j,k)) * stoch_CS%sppt_wts(i,j)
       tv%S(i,j,k) = max(s_in(i,j,k) + (tv%S(i,j,k)-s_in(i,j,k)) * stoch_CS%sppt_wts(i,j), 0.0)
+      tv%T(i,j,k) = max(t_in(i,j,k) + (tv%T(i,j,k)-t_in(i,j,k)) * stoch_CS%sppt_wts(i,j), -0.054*tv%S(i,j,k))
```

Comment:
This line of code includes a hard-coded dimensional constant, 0.054 degC/ppt, and will not pass the dimensional consistency testing.  (It is missing a factor of US%S_to_ppt*US%degC_to_C.) Moreover, this is a hard-coded form that is specific to a particular (simple) expression for the freezing point.  More general expressions might be nonlinear on include a pressure dependence so that they can be appropriately used under thick ice.

  The strongly preferred approach would be to use a call to calculate_Tfreeze for this limitation to avoid sub-freezing temperatures, as is done in the routine make_frazil() in MOM_diabatic_aux.F90.

Reply:
@Hallberg-NOAA I just updated the branch with Phil's modification, can you check that again ? thanks